jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -195,6 +187,28 @@ protected void doStart() throws Exception {
     }
 
     super.doStart();
+
+    // Look for a datastore service within the context
+    ClassLoader orig = Thread.currentThread().getContextClassLoader();
```

Comment:
Containers adding filters that are loaded from their classpath is not unusual,  but containers adding filters that are loaded from the webapps classpath is.

It is a rather strange behaviour really, the container is saying that the webapp must provide its own Datastore implementation, but then intervenes in that impl to impose a transaction rollback policy.     It is a strange halfway house as I would expect that such a container policy would be appropriate to enforce on a container provided services, but not on a service that the container is insisting the webapp provides for itself.

I better solution would be for the jar that provides the datastore service to also contain a utility filter for enforcing a rollback policy.  Applications could then choose to instantiate it or not, or it would be easier for the container to put the filter on there classpath if so desired.